custom_components/rpi_gpio/hub.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+from . import DOMAIN
+
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP, EVENT_HOMEASSISTANT_START
+from homeassistant.exceptions import IntegrationError
+
+from typing import Dict
+from datetime import timedelta
+import gpiod
+
+from gpiod.line import Direction, Value, Bias, Drive, Edge, Clock
+EventType = gpiod.EdgeEvent.Type
+
+BIAS = { 
+    "UP": Bias.PULL_UP, 
+    "DOWN": Bias.PULL_DOWN,
+    "DISABLED": Bias.DISABLED,
+    "AS_IS": Bias.AS_IS,
+}
+DRIVE = { 
+    "OPEN_DRAIN": Drive.OPEN_DRAIN, 
+    "OPEN_SOURCE": Drive.OPEN_SOURCE, 
+    "PUSH_PULL": Drive.PUSH_PULL, 
+} 
+
+class Hub:
+
+    def __init__(self, hass: HomeAssistant, path: str) -> None:
+        """GPIOD Hub"""
+
+        self._path = path
+        self._chip :  gpiod.Chip
+        self._name = path
+        self._id = path
+        self._hass = hass
+        self._online = False
+        self._lines : gpiod.LineRequest = None
+        self._config : Dict[int, gpiod.LineSettings] = {}
+        self._edge_events = False
+        self._entities = {}
+
+        if path:
+            # use config
+            if self.verify_gpiochip(path):
+                self._online = True
+                self._path = path
+        else:
+            # discover
+            for d in [0,4,1,2,3,5]:
+                # rpi3,4 using 0. rpi5 using 4
+                path = f"/dev/gpiochip{d}"
+                if self.verify_gpiochip(path):
+                    self._online = True
+                    self._path = path
+                    break
+
+        if not self._online:
+            _LOGGER.error("No gpio device detected, bailing out")
+            raise IntegrationError("No gpio device detected")
```

Comment:
```suggestion
            raise HomeAssistantError("No gpio device detected")
```
`IntegrationError` is internal for HA, integrations should raise `HomeAssistantError`

Reply:
fixed.